src/FakeItEasy/Core/DefaultFakeObjectCallFormatter.cs

```diff
@@ -184,20 +183,20 @@ private string GetArgumentValueAsString(object? argumentValue)
         private void AppendArguments(StringBuilder builder, ArgumentValueInfo[] arguments)
         {
             var totalNumberOfArguments = arguments.Length;
-            var argumentIndex = 0;
             foreach (var argument in arguments)
             {
-                AppendArgumentSeparator(builder, argumentIndex, totalNumberOfArguments);
-                this.AppendArgumentValue(builder, argumentIndex, argument);
-                argumentIndex++;
+                AppendArgumentSeparator(builder, argument.ArgumentIndex, totalNumberOfArguments);
+                this.AppendArgumentValue(builder, argument);
             }
         }
 
         private struct ArgumentValueInfo
         {
-            public object? ArgumentValue { get; set; }
+            public int ArgumentIndex { get; init; }

```

Comment:
Looks like we're missing something to be able to use `init`. I think we can just declare `IsExternalInit` ourselves for TFMs that don't have it.

Or maybe we could use a record? (but again, we'll probably need to declare some missing types ourselves)

(and of course, the simplest fix is probably to go back to using a normal setter instead of `init`; it doesn't really matter, the type is private anyway.)